### PR TITLE
*: support the binary plan through EXPLAIN format='binary' and tidb_decode_binary_plan

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/35889

### What is changed and how it works?

1. Support generate the binary plan. (common_plans.go)
2. Add EXPLAIN [ANALYZE] format = 'binary' to output the binary plan. (explain_format.go, common_plans.go)
3. Add tidb_decode_binary_plan() to decode the binary plan and get the result almost the same as EXPLAIN [ANALYZE]. (binary_plan_decode.go, expression/*, functions.go)
4. Update protobuf definition and some minor fixes for the previous flat plan and access object code. (flat_plan.go, access_object.go)

More unit tests for `tidb_decode_binary_plan` would be in the next PR.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by AilinKid** [comment]
ref: https://github.com/pingcap/tipb/pull/269

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Review by winoros — APPROVED** [review]

File: planner/core/common_plans.go [hunk 1]
```diff
@@ -1412,6 +1420,145 @@ func (e *Explain) getOperatorInfo(p Plan, id string) (string, string, string, st
...
+	// Please see comments in EncodeFlatPlan() for this case.
+	// We keep consistency with EncodeFlatPlan() here.
+	if flat.InExecute {
+		return nil
+	}
+	res := &tipb.ExplainData{}
+	for _, op := range flat.Main {
+		rootStats, copStats, _, _ := getRuntimeInfo(explainCtx, op.Origin, nil)
+		if rootStats != nil || copStats != nil {
+			res.WithRuntimeStats = true
```
[the remark this answers is not in the document]
Comment (time-and-fate) [reply planner/core/common_plans.go:1450]: The `WithRuntimeStats` is a global property for the binary plan tree, not for a single operator. So I think it's better to set it here.

[the remark this answers is not in the document]
Comment (time-and-fate) [reply planner/core/common_plans.go:1451]: I added a comment to clarify this.

[the remark this answers is not in the document]
Comment (AilinKid) [reply planner/core/common_plans.go:1451]: Good.

**Pull request merged**